tests/qa_automation/prepare_qa_repo.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: the step to prepare QA:Head repository
+# Maintainer: Yong Sun <yosun@suse.com>
+
+use strict;
+use testapi;
+use base "opensusebasetest";
+use utils;
+
+# Add qa head repo for kernel testing. If QA_SERVER_REPO is set,
+# remove all existing zypper repos first
+sub prepare_repos {
+    my $self           = shift;
+    my $qa_head_repo = get_var('QA_HEAD_REPO', '');
+    my $qa_web_repo  = get_var('QA_WEB_REPO',  '');
+    unless ($qa_head_repo) {
+        die "No QA_HEAD_REPO specified!";
+    }
```

Comment:
same as above to use get_required_var
https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L599

Reply:
@mitiao @cachen  `get_required_var` only for *QA_HEAD_REPO* .. second var is only optional so `get_var` is right

Reply:
Both QA_HEAD_REPO and QA_TESTSUITE are need, they good to use get_required_var. QA_WEB_REPO and QA_VERSION are options for special testsuite, they can use get_var.

Good to learn from your both, thanks a lot :)